packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -105,8 +125,54 @@ export const AlephiumConnectProvider: React.FC<{ children?: React.ReactNode }> =
 
   const [account, setAccount] = useState<Account & { network: NetworkId }>()
   const [signerProvider, setSignerProvider] = useState<SignerProvider | undefined>()
+  const [balance, setBalance] = useState<node.Balance | undefined>()
+
+  const updateBalance = useCallback(async () => {
+    const nodeProvider = tryGetNodeProvider() ?? signerProvider?.nodeProvider
+    if (nodeProvider && account) {
+      const newBalance = await nodeProvider.addresses.getAddressesAddressBalance(account.address)
+      setBalance((prevBalance) => {
+        if (prevBalance !== undefined && isBalanceEqual(prevBalance, newBalance)) {
+          return prevBalance
+        }
+        return newBalance
+      })
+    } else if (account === undefined) {
+      setBalance(undefined)
+    }
+  }, [signerProvider?.nodeProvider, account, setBalance])
+
+  const updateBalanceForTx = useCallback(
+    (txId: string, confirmations?: number) => {
+      const expectedConfirmations = confirmations ?? 1
+      const pollingInterval = account?.network === 'devnet' ? 1000 : 4000
+      const messageCallback = async (txStatus: node.TxStatus): Promise<void> => {
+        if (txStatus.type === 'Confirmed' && (txStatus as node.Confirmed).chainConfirmations >= expectedConfirmations) {
+          await updateBalance()
+        }
+      }
+      const errorCallback = (err: any, subscription: Subscription<node.TxStatus>): Promise<void> => {
+        subscription.unsubscribe()
+        console.error(`tx status subscription error: ${err}`)
+        return Promise.resolve()
+      }
+      const options: SubscribeOptions<node.TxStatus> = {
+        pollingInterval,
+        messageCallback,
+        errorCallback
+      }
+      subscribeToTxStatus(options, txId, undefined, undefined, expectedConfirmations)
+    },
+    [updateBalance, account?.network]
+  )
+
+  useEffect(() => {
+    updateBalance()
+  }, [updateBalance])
```

Comment:
This would need a balance query always, which is not necessary for all apps. Some apps might just want to get the account information.

Recommend:
1. Introduce another context for balance, which is `undefined` by default
2. `useBalance` would trigger the update the balance in the context if the balance is `undefined`, otherwise it will skip
3. `updateBalanceForTx` still works the same

Reply:
Sounds great, will fix it, thanks!